src/runtime_src/core/tools/common/ReportDynamicRegion.cpp

```diff
@@ -268,49 +284,57 @@ ReportCu::writeReport( const xrt_core::device* /*_pDevice*/,
   boost::format cuFmt("    %-8s%-30s%-16s%-8s%-8s\n");
 
   //check if a valid CU report is generated
-  const boost::property_tree::ptree& pt_cu = _pt.get_child("compute_units", empty_ptree);
-  if(pt_cu.empty())
+  const boost::property_tree::ptree& pt_dfx = _pt.get_child("dynamic_regions", empty_ptree);
+  if(pt_dfx.empty())
     return;
 
-  _output << "Compute Units" << std::endl;
-  _output << "  PL Compute Units" << std::endl;
-  _output << cuFmt % "Index" % "Name" % "Base_Address" % "Usage" % "Status";
-  try {
-    int index = 0;
-    for(auto& kv : pt_cu) {
-      const boost::property_tree::ptree& cu = kv.second;
-      if(cu.get<std::string>("type").compare("PL") != 0)
-        continue;
-      std::string cu_status = cu.get_child("status").get<std::string>("bit_mask");
-      uint32_t status_val = std::stoul(cu_status, nullptr, 16);
-      _output << cuFmt % index++ %
-	      cu.get<std::string>("name") % cu.get<std::string>("base_address") %
-	      cu.get<std::string>("usage") % xrt_core::utils::parse_cu_status(status_val);
+  for(auto& k_dfx : pt_dfx) {
+    const boost::property_tree::ptree& dfx = k_dfx.second;
+    _output << "Xclbin UUID" << std::endl;
+    _output << "  " + dfx.get<std::string>("xclbin_uuid", "N/A") << std::endl;
+    _output << std::endl;
+
+    const boost::property_tree::ptree& pt_cu = dfx.get_child("compute_units", empty_ptree);
+    _output << "Compute Units" << std::endl;
+    _output << "  PL Compute Units" << std::endl;
+    _output << cuFmt % "Index" % "Name" % "Base_Address" % "Usage" % "Status";
+    try {
+      int index = 0;
+      for(auto& kv : pt_cu) {

```

Comment:
This for loop is for ranges and yet the code wishes to treat it as an array.  For future implementations, you might wish to consider the following:
```
for (std::size_t index = 0; index < pt_cu.size(); ++index) {
  auto & kv = pt_cu[index];
...
```
Makes the intent and code a little easier to read.

And yes, I know it goes against the coding guidelines: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Res-for-range

Side Note
If we were using C++20, then we could potentially use:
`for( unsigned int count = 0; auto& kv : pt_cu)`